src/LowerTriangularMatrices/lower_triangular_matrix.jl

```diff
@@ -162,127 +239,130 @@ function Base.copyto!(  L1::LowerTriangularMatrix{T},   # copy to L1
+    M[lower_triangle_indices] = convert.(T, L.data)
 
-        for l in m:lmax         # convert and copy for lower triangle
-            lm += 1
-            M[l, m] = convert(T, M[lm])
-        end
-    end
     M
 end
 
-function LowerTriangularMatrix{T}(M::LowerTriangularMatrix) where T
-    L = LowerTriangularMatrix{T}(undef, size(M)...)
+function LowerTriangularMatrix{T}(M::LowerTriangularMatrix{T2,ArrayType}) where {T,T2,ArrayType}
+    L = LowerTriangularMatrix{T,ArrayType}(undef, size(M)...)
     copyto!(L, M)
     return L
 end
 
-LowerTriangularMatrix(M::LowerTriangularMatrix{T}) where T = LowerTriangularMatrix{T}(M)
-
-function Base.convert(::Type{LowerTriangularMatrix{T1}}, L::LowerTriangularMatrix{T2}) where {T1, T2}
-    return LowerTriangularMatrix{T1}(L.data, L.m, L.n)
+function Base.convert(::Type{LowerTriangularArray{T1,N,ArrayTypeT1}}, L::LowerTriangularArray{T2,N,ArrayTypeT2}) where {T1, T2, N, ArrayTypeT1<:AbstractArray{T1}, ArrayTypeT2<:AbstractArray{T2}}
+    return LowerTriangularArray{T1,N,ArrayTypeT1}(L.data, L.m, L.n)
 end
 
-function Base.similar(::LowerTriangularMatrix{T}, size::Integer...) where T
-    return LowerTriangularMatrix{T}(undef, size...)
+function Base.convert(::Type{LowerTriangularMatrix{T}}, L::LowerTriangularMatrix) where T
+    return LowerTriangularMatrix{T,Vector{T}}(L.data, L.m, L.n)
 end
 
-function Base.similar(::LowerTriangularMatrix{T}, size::NTuple{2, Integer}) where T
-    return LowerTriangularMatrix{T}(undef, size...)
+function Base.similar(::LowerTriangularArray{T,N,ArrayType}, size::S) where {T, N, ArrayType, S<:Tuple}
+    return LowerTriangularArray{T,N,ArrayType}(undef, size...)
 end
 
-function Base.similar(L::LowerTriangularMatrix, ::Type{T}) where T
-    return LowerTriangularMatrix{T}(undef, size(L)...)
+function Base.similar(L::LowerTriangularArray{S,N,ArrayType}, ::Type{T}) where {T, S, N, ArrayType}
+    new_array_type = typeof(T.(L.data)) # TODO: not sure how else to infer this type 
+    return LowerTriangularArray{T,N,new_array_type}(undef, size(L)...)
 end
 
-Base.similar(L::LowerTriangularMatrix{T}) where T = similar(L, T)
-
+Base.similar(L::LowerTriangularArray{T}) where T = similar(L, T)
+ 
 # ARITHMETIC
 # only mul/div with scalar and addition/subtraction, others are converted to Matrix
-function Base.:(*)(L::LowerTriangularMatrix{T}, s::Number) where T
-    M = similar(L)
+function Base.:(*)(L::LowerTriangularArray{T}, s::Number) where T
     sT = convert(T, s)
-    @inbounds for i in eachindex(L, M)
-        M[i] = L[i] * sT
-    end
-    M
+    LowerTriangularArray(L.data .* sT, L.m, L.n)
 end
 
-Base.:(*)(s::Number, L::LowerTriangularMatrix) = L*s         # commutative
-Base.:(/)(L::LowerTriangularMatrix, s::Number) = L*inv(s)
-Base.:(/)(s::Number, L::LowerTriangularMatrix) = L/s
+Base.:(*)(s::Number, L::LowerTriangularArray) = L*s         # commutative
+Base.:(/)(L::LowerTriangularArray, s::Number) = L*inv(s)
+Base.:(/)(s::Number, L::LowerTriangularArray) = L/s
```

Comment:
ehhh what? s/L = L/s ?

Reply:
Haha, yeah, right that makes no sense. This bit I just renamed though, that's in the current version as well ;)

Reply:
I'm glad we don't use it 😅 or are we? 🙈